cesium/internal/unary/iterator.go

```diff
@@ -138,7 +138,7 @@ func (i *Iterator) Next(ctx context.Context, span telem.TimeSpan) (ok bool) {
 
 	i.reset(i.view.End.SpanRange(span).BoundBy(i.bounds))
 
-	if i.view.IsZero() {
+	if i.view.IsZero() || i.view.End.BeforeEq(i.internal.TimeRange().Start) {
```

Comment:
I'd be interested in getting an explanation of why you added this

Reply:
If the entire view is before the start of the internal domain, that means there's no data to be read.

Added to address a bug where if the view is put between two domains via `Next`, it unnecessarily moves on to the next domain and skip the previous one. (domain 1: [10, 20), domain 2: [30, 40), domain 3: [50, 60), say view is at [23, 24), the iterator position is at domain 2 – instead of reading nothing and doing nothing, the iterator actually moved on to domain 3, skipping domain 2 entirely.